### PR TITLE
Cleanup and additions

### PR DESCRIPTION
**Pull request by ghost**
## Description
Too much lol...
- create some hooks
- cleanup the routing
- fix some stuff on login page
- reorganize some files
- adjust the navigation
- update the course and user states slightly

### Special Notes
Some cool stuff in here and more functionality 🎉 

### How has this been tested?
I tried my best to test these scenarios locally

## Requirements
Fill out the checkboxes below, verifying that all the requirements have been satisfied:
- [x] Have I documented where necessary (technical docs, how to docs, etc.)?
- [x] Have I requested at least 1 reviewer?
- [x] Have I left comments so my work is easy to understand?
- [x] Have I looked over my code again to weed out any mistakes I may have made or details missed?
- [x] Does my work reflect all the needs of this given task?

### CODE REVIEW

File: src/App.js
Comment (gavenrobertson) [inline src/App.js:1]: Looks Great! Thank you for cleaning this up! Should help a lot when referring to Routes!

File: src/components/nav/SideNavbar.js
Comment (gavenrobertson) [inline src/components/nav/SideNavbar.js:1]: Like this implementation of the sidebar creation🙌🙂

File: src/hooks/useCourses.js
Comment (gavenrobertson) [inline src/hooks/useCourses.js:1]: Just a quick question! Is this essentially checking if there are courses, if no courses exist then you call the GET request? I assume it works great, just wanted to get a better understanding!

**Review by gavenrobertson — APPROVED** [review]
Looks Great Matthew! A lot of new additions as well as quality of life adjustments! Looks Good 👍

File: src/pages/Landing.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                })}
+            </div>}
+
+            {courses.teacherCourses && <div id="teacher-courses">
+                <div className="buttons">
+                    <Button variant="secondary">Add Course</Button>
+                </div>
+                <div>
+                    {courses.teacherCourses.map((teacherCourse) => {
+                        return <CourseCard key={teacherCourse.id} course={teacherCourse} role={"landing"} />
```
Comment (METrimble) [inline src/pages/Landing.js:28]: is the role here supposed to be landing? or is it teacher?

File: src/components/CourseCard.js [hunk 2]
```diff
@@ -47,23 +38,18 @@ function CourseCard (props) {
...
-              {props.view==="student" ? 
-              <Link to={`/singleCourse`}>
-                <Button className="hideBtn" onClick={() => {saveCourseClicked(props.currentCourse)}}>
-                  View Course
-                </Button>
-              </Link> : 
-              <Link to={`/singleCourse`}>
+               
+              <Link to={`/${props.course.id}`}>
                 <Button className="hideBtn" onClick={() => {saveCourseClicked(props.currentCourse)}}>
```
Comment (METrimble) [inline src/components/CourseCard.js:49]: Does this still go to a saveCourseClicked Function? I can't seem to find where this onclick is supposed to go now that the function is removed

**Review by METrimble — COMMENTED** [review]
You're super human. I added a few questions for clarification, but thank you for this. Good night!

File: src/pages/Landing.js (hunk 1 above)
Comment (mitchellstewart7) [reply src/pages/Landing.js:28]: Good catch, I believe this should be teacher

**Review by mitchellstewart7 — APPROVED** [review]
Looks good to me. Thank you for removing the stuff the old team did.

**Pull request merged**